test.js

```diff
@@ -94,36 +83,48 @@ describe('cli', function () {
   describe('with mocked git', function () {
     it('--sign signs the commit and tag', function () {
       // mock git with file that writes args to gitcapture.log
-      mockGit('require("fs").appendFileSync("gitcapture.log", JSON.stringify(process.argv.splice(2)) + "\\n", "utf8")')
-      writePackageJson('1.0.0')
+      mockGit('require("fs").appendFileSync("gitcapture.log", JSON.stringify(process.argv.splice(2)) + "\\n")')
```

Comment:
> I wonder if the mocking library could be made more generic, as a tool for mocking an arbitrary bin in path (then you could instantiate one for git).

I have created [mock-bin](https://github.com/stevemao/mock-bin) and mock-git is depending on it.

> I wonder if we could provide a few common "logic scripts"

There is an optional arg, [command](https://github.com/stevemao/mock-git#mockgitjs-command), you could specify. The edge case I mentioned is when your command is `commit`, it won't affect other commands such as `push`. Although it doesn't matter at the moment here.